pkg/ds/daemon_set.go

```diff
@@ -25,6 +25,14 @@ const (
 type DaemonSet interface {
 	ID() fields.ID
 
+	IsDisabled() bool
+
+	PodID() types.PodID
+
+	GetNodeSelector() klabels.Selector
+
+	EligibleNodes() ([]types.NodeName, error)
+
 	WatchDesires(
```

Comment:
nit: usually we write comments documenting the functions on an interface, at least the non-obvious ones. Probably `EligibleNodes()` and `WatchDesires()` at least are good candidates for that